### PR TITLE
fix: sed works with slashes in regex

### PR DESCRIPTION
**Pull request by nfischer**
This should make `shx sed` work a little bit better. Also, this should make the code a bit more readable. If you have any readability suggestions, chime in and I'll implement those.

### CODE REVIEW

File: src/shx.js [hunk 1]
```diff
@@ -37,10 +37,13 @@ export const shx = (argv) => {
     newArgs = [];
     let lookingForSubstString = true;
     args.forEach((arg) => {
-      const match = arg.match(/^s\/(.*)\/(.*)\/(g?)$/);
+      const match = arg.match(/^s\/(.*[^\\])\/(.*[^\\])\/(g?)$/);
```
Comment (ariporad) [inline src/shx.js]: Shouldn't this be `/^s\/([^\\.]*)\/([^\\.]*)\/(g?)$/`;

Comment (nfischer) [inline src/shx.js]: @ariporad A sed-regex is allowed to have backslashes in it, but it's not allowed to end in a backslash. The end of a sed-regex is designated by the first appearance of a non-escaped `/`. If you want to include a `/` in the regex, you would use `\/`. I think your JS-regex would prevent that from working.

Comment (ariporad) [inline src/shx.js]: Ok.

**Comment by ariporad** [comment]
LGTM!

**Comment by nfischer** [comment]
LGTM

**Pull request merged**